### DISCUSSION TITLE
Using Caddy for Reverse Proxy

### DISCUSSION THREAD

**Discussion by Aniket-IN** [highlight/highlight#9347] [General]
Hi Team, 

Thank you so much for this amazing software. 
I was trying out the Hobby deployment, it got installed without any issue.

However, now I'm trying to put it behind Reverse Proxy, on the docs it has a example of using Traefik as reverse proxy,
https://www.highlight.io/docs/getting-started/self-host/traefik-proxy
But I use Caddy for self hosting apps on my homelab. 

Is there anyway I can use Caddy instead? 

I'm a bit confused, how the reverse proxy is working with Traefik, it seems like both the backend and frontend are on same host `highlight.example.com`

How does Traefik know, where to route each request (backend or frontend) 🤔

**Comment by Vadman97** [comment]
Thanks for the question, @Aniket-IN ! In general, you can have a proxy separate the backend and frontend traffic by port or by path. In the docs, we describe separating it by port (so that the frontend and backend live on different ports), where the listening port is configurable via the `docker/.env` file. Another way to separate the traffic would be by path (since we use graphql, configuring the proxy to send app `/public` and `/private` traffic to the backend, and all other traffic to the frontend container). 

Let me know if you get stuck trying to get Caddy or Traefik set up!

**Reply by Aniket-IN** [reply]
Thanks you so much for the explanation @Vadman97 
Now the dashboard is working 🎉

This is the Caddy config I used (in case someone reads this in future)
```Caddyfile
handle /public {
    reverse_proxy "http://highlight-vm-ip:8082"
}

handle /public/* {
    reverse_proxy "http://highlight-vm-ip:8082"
}

handle /private {
    reverse_proxy "http://highlight-vm-ip:8082"
}

handle /private/* {
    reverse_proxy "http://highlight-vm-ip:8082"
}

# Reverse proxy rest of requests to frontend
reverse_proxy "http://highlight-vm-ip:3000"
```

However, I'm having CORS issue. 
```sh
Access to XMLHttpRequest at 'https://highlight.mydomain.com/' from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```
Is there any `.env` change or something that I have to do? 🤔
On my Nextjs project, I'm initializing Highlight SDK like this:
```tsx
      <HighlightInit
        backendUrl="https://highlight.mydomain.com"
        projectId={"1jdkoe52"}
        serviceName="my-nextjs-frontend"
        tracingOrigins
        networkRecording={{
          enabled: true,
          recordHeadersAndBody: true,
          urlBlocklist: [],
        }}
      />
```

**Reply by Aniket-IN** [reply]
Oops may bad, after searching this for hours, I figured out the reason. 
Basically the backendUrl needs to be set to `https://highlight.mydomain.com/public` I was using just `https://highlight.mydomain.com` instead. 

Now it's working perfectly fine, thank you so much @Vadman97 for all the help 👏
